repos/system_upgrade/common/actors/distributionsignedrpmscanner/actor.py

```diff
@@ -0,0 +1,94 @@
+import json
+import os
+
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common import rhui
+from leapp.libraries.common.config import get_env
+from leapp.libraries.stdlib import api
+from leapp.models import DistributionSignedRPM, InstalledRedHatSignedRPM, InstalledRPM, InstalledUnsignedRPM
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+from leapp.utils.deprecation import suppress_deprecation
+
+
+@suppress_deprecation(InstalledRedHatSignedRPM)
+class DistributionSignedRpmScanner(Actor):
+    """Provide data about installed RPM Packages signed by the distribution.
+
+    After filtering the list of installed RPM packages by signature, a message
+    with relevant data will be produced.
+    """
+
+    name = 'distribution_signed_rpm_scanner'
+    consumes = (InstalledRPM,)
+    produces = (DistributionSignedRPM, InstalledRedHatSignedRPM, InstalledUnsignedRPM,)
+    tags = (IPUWorkflowTag, FactsPhaseTag)
+
+    def process(self):
+        # TODO(pstodulk): refactor this function
+        # - move it to the private library
+        # - split it into several functions (so the main function stays small)
+        # FIXME(pstodulk): gpg-pubkey is handled wrong; it's not a real package
+        # and create FP report about unsigned RPMs. Keeping the fix for later.
+        distribution = self.configuration.os_release.release_id
```

Comment:
keeping it for a followup PR, as this is the way how the actor was implemented originally. having a few other lines in it, does not make it worse - and moving only this part to the library does not help neither. From my pov, this actor needs to be refactored completely and to reduce amount of changes and make it more clear for review, I kept this as it is. PR with the refactoring will be then more clean.

Reply:
Are we in a rush to get this in?

Reply:
We discussed that on a mtg, preferred is to make it early than later. One of reasons why we want to do it now is that we are not in less hectic time (when I do not speak about number of investigations I had to work on in last weeks), and this PR touches 100files right now. Also we have small amount of PR opened - we can expect very different situations when people start to work on RHEL 10. Another reason is that I would create still the downstream build, and I prefer to do it rather tomorrow than on my last wday in this year (in such a case I would prefer to move it already to Jan). Also I am not sure what I will have to work on in upcoming days regarding. I was thinking to refactor it already in a new commit, but I really think it will be better to separate it.

Reply:
Of course, if upstream tests are operational again. working on the refactoring now.

Reply:
@fernflower done! On top of that, I've resolved also the fixme about the gpg-pubkey (I couldn't hold my fingers..). I was thinking whether to mark gpg-pubkey at all as signed rpms, as it's not a real rpm, but in the end, as we have it between InstalledRPMs already, it make sense to have them somewhere.

Not sure whether this approach is better, or whether we should keep sorting them between signed and unsigned packages to check for imported keys that are from the distro itself. In the second case, it would need to be fixed the actor with FP report. But somehow I am seeing this approach better. Let me know if you have a different idea about that change.

@oamg/developers the "fixme" note can be discussed widely. Checked with QE, it seems we do not have currently  any tests for that, so it should be ok. Only tests that are related are connected to check of GPG signatures, which is unaffected by this change.